### PR TITLE
[BE][CI] migrate .jenkins to .ci

### PR DESCRIPTION
**Pull request by janeyx99**
PLEASE LOOK AT https://github.com/pytorch/pytorch/pull/92621 INSTEAD
-----

Completes what I started in https://github.com/pytorch/pytorch/pull/82753 

Most the changes here are:
- Copying .jenkins files to .ci
- Replacing .jenkins references with .ci

Once this lands, we'll likely have to tell people to rebase past this change.

Next steps:
- Remove .jenkins
- Remove old things in .ci

Why don't I remove .jenkins already? 
1. Removing it makes the bc test not pass and starts a rabbit hole of problems as can be seen in the first PR attempt.
2. There may be other dependencies from other repos to .jenkins and we may want a buffer time/not have this PR grow in scope unnecessarily.

### CODE REVIEW

File: .ci/pytorch/test.sh [hunk 1]
```diff
@@ -0,0 +1,906 @@
...
+  pip install -r requirements.txt
+    # TODO: in a few weeks, once most commits adopt .ci over .jenkins, remove the fallback
+  COMMON_BUILD_SCRIPT="$(dirname "${BASH_SOURCE[0]}")/common-build.sh"
+  if [[ -f "$COMMON_BUILD_SCRIPT" ]]; then
+    # shellcheck source=./common-build.sh
+    bash "$COMMON_BUILD_SCRIPT"
+  else
+    # shellcheck source=./common-build.sh
+    bash .jenkins/pytorch/common-build.sh
+  fi
```
Comment (janeyx99) [inline .ci/pytorch/test.sh:676]: i changed only these manually.

all other changes are mostly a .jenkins -> .ci

**Review by huydhn — APPROVED** [review]
LGTM!  Should we look around for a solution to keep `.jenkins` and `.ci` in sync in the interim before the former can be removed?

**Comment by janeyx99** [comment]
> LGTM! Should we look around for a solution to keep `.jenkins` and `.ci` in sync in the interim before the former can be removed?

Maybe? But it may not be very worth it since I intend for that interim to be small (a week at most).

Another thing we could do to completely eliminate this gap is to include the .jenkins removal here, and force merge the bc test breaking as an intentional break.

**Comment by huydhn** [comment]
> Maybe? But it may not be very worth it since I intend for that interim to be small (a week at most).
> 
> Another thing we could do to completely eliminate this gap is to include the .jenkins removal here, and force merge the bc test breaking as an intentional break.

That sounds good to me. I like your approach here better with `.jenkins` removal done in a separate PR.

File: .ci/caffe2/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Jenkins
```
Comment (ZainRizvi) [inline .ci/caffe2/README.md:1]: Want to rename the jenkins references in this file?

File: .ci/caffe2/test.sh [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+  # Currently these tests are failing on ROCM platform:
+
+  # On ROCm, RCCL (distributed) development isn't complete.
+  # https://github.com/ROCmSoftwarePlatform/rccl
+  rocm_ignore_test+=("--ignore $caffe2_pypath/python/data_parallel_model_test.py")
+
+  # This test has been flaky in ROCm CI (but note the tests are
+  # cpu-only so should be unrelated to ROCm)
+  rocm_ignore_test+=("--ignore $caffe2_pypath/python/operator_test/blobs_queue_db_test.py")
+  # This test is skipped on Jenkins(compiled without MKL) and otherwise known flaky
```
Comment (ZainRizvi) [inline .ci/caffe2/test.sh:131]: more jenkins refs

File: .ci/caffe2/test.sh [hunk 4]
```diff
@@ -0,0 +1,172 @@
+#!/bin/bash
+
+# shellcheck source=./common.sh
+source "$(dirname "${BASH_SOURCE[0]}")/common.sh"
+
+if [[ ${BUILD_ENVIRONMENT} == *onnx* ]]; then
+  pip install click mock tabulate networkx==2.0
+  pip -q install --user "file:///var/lib/jenkins/workspace/third_party/onnx#egg=onnx"
```
Comment (ZainRizvi) [inline .ci/caffe2/test.sh:8]: This prob needs to go too...

File: .ci/onnx/README.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# Jenkins
```
Comment (ZainRizvi) [inline .ci/onnx/README.md:1]: here

File: .ci/onnx/test.sh [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+if [[ "$BUILD_ENVIRONMENT" == *onnx* ]]; then
+  pip install -q --user --no-use-pep517 "git+https://github.com/pytorch/vision.git@$(cat .github/ci_commit_pins/vision.txt)"
+  pip install -q --user ninja flatbuffers==2.0 numpy==1.21.5 onnxruntime==1.12.1 beartype==0.10.4 onnx==1.12.0
+  # TODO: change this when onnx-script is on testPypi
+  pip install 'onnx-script @ git+https://github.com/microsoft/onnx-script@4f3ff0d806d0d0f30cecdfd3e8b094b1e492d44a'
+  # numba requires numpy <= 1.20, onnxruntime requires numpy >= 1.21.
+  # We don't actually need it for our tests, but it's imported if it's present, so uninstall.
+  pip uninstall -q --yes numba
+  # JIT C++ extensions require ninja, so put it into PATH.
+  export PATH="/var/lib/jenkins/.local/bin:$PATH"
```
Comment (ZainRizvi) [inline .ci/onnx/test.sh:69]: And this I'm guessing

**Review by ZainRizvi — COMMENTED** [review]
There's a bunch of jenkins references still existing in the copied files.  Are those supposed to still be there?

**Review by malfet — APPROVED** [review]
Please consider creating symlinks and deleting them later to make transition process smoother.

**Comment by janeyx99** [comment]
Closing in favor of https://github.com/pytorch/pytorch/pull/92621.

@ZainRizvi  I think it'd be better as a followup to replace the Jenkins refs (since those are mostly doc changes)

**Pull request closed without merging**